### ISSUE TITLE
Metallb drops traffic to services periodically Failed to resolve : lookup : no such host

### ISSUE THREAD

**Issue by alphabet5** [metallb/metallb#1533]
This seems related to #1507 perhaps? I was seeing the same thing on 0.13.3 except a lot more frequent disconnects.

Metallb is installed via Flux/Helm

```yaml
---
apiVersion: source.toolkit.fluxcd.io/v1beta1
kind: HelmRepository
metadata:
  name: metallb
  namespace: flux-system
spec:
  interval: 1h0m0s
  url: https://metallb.github.io/metallb

---
apiVersion: helm.toolkit.fluxcd.io/v2beta1
kind: HelmRelease
metadata:
  name: metallb
  namespace: flux-system
spec:
  chart:
    spec:
      chart: metallb
      sourceRef:
        kind: HelmRepository
        name: metallb
      version: '0.13.4'
  install:
    crds: CreateReplace
    remediation:
      retries: 3
    createNamespace: true
  interval: 1h0m0s
  targetNamespace: metallb-system
  upgrade:
    crds: CreateReplace
    remediation:
      retries: 3
      remediateLastFailure: true
  values:
    fullnameOverride: metallb
```

Metallb operator manifests installed from github (double checked it was up to date yesterday.)

Metallb crds

```
---
apiVersion: metallb.io/v1beta1
kind: MetalLB
metadata:
  name: metallb
  namespace: metallb-system
---
apiVersion: metallb.io/v1beta1
kind: IPAddressPool
metadata:
  name: default
  namespace: metallb-system
spec:
  addresses:
    - 10.5.54.24/32
    - 10.5.54.25/32
    - 10.5.54.26/32
---
apiVersion: metallb.io/v1beta1
kind: L2Advertisement
metadata:
  name: default
  namespace: metallb-system
spec:
  ipAddressPools:
  - default
```

```
2022-07-21T23:19:41.108374436Z {"level":"error","ts":1658445581.1080754,"logger":"cert-rotation","msg":"Error updating webhook with certificate","name":"bgppeers.metallb.io","gvk":"apiextensions.k8s.io/v1, Kind=CustomResourceDefinition","error":"Operation cannot be fulfilled on customresourcedefinitions.apiextensions.k8s.io \"bgppeers.metallb.io\": the object has been modified; please apply your changes to the latest version and try again","stacktrace":"github.com/open-policy-agent/cert-controller/pkg/rotator.(*ReconcileWH).Reconcile\n\t/go/pkg/mod/github.com/open-policy-agent/cert-controller@v0.3.0/pkg/rotator/rotator.go:642\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.2/pkg/internal/controller/controller.go:121\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.2/pkg/internal/controller/controller.go:320\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.2/pkg/internal/controller/controller.go:273\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.2/pkg/internal/controller/controller.go:234"}
2022-07-21T23:19:41.109881161Z {"level":"error","ts":1658445581.1081822,"msg":"Reconciler error","controller":"cert-rotator","object":{"name":"webhook-server-cert","namespace":"metallb-system"},"namespace":"metallb-system","name":"webhook-server-cert","reconcileID":"555aa692-2685-42aa-9242-b6696b2914c3","error":"Operation cannot be fulfilled on customresourcedefinitions.apiextensions.k8s.io \"bgppeers.metallb.io\": the object has been modified; please apply your changes to the latest version and try again","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.2/pkg/internal/controller/controller.go:273\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.2/pkg/internal/controller/controller.go:234"}
2022-07-21T23:19:41.109900447Z {"level":"info","ts":1658445581.1087449,"logger":"cert-rotation","msg":"Ensuring CA cert","name":"metallb-webhook-configuration","gvk":"admissionregistration.k8s.io/v1, Kind=ValidatingWebhookConfiguration","name":"metallb-webhook-configuration","gvk":"admissionregistration.k8s.io/v1, Kind=ValidatingWebhookConfiguration"}
2022-07-21T23:19:41.187261470Z {"level":"info","ts":1658445581.1860316,"logger":"cert-rotation","msg":"Ensuring CA cert","name":"addresspools.metallb.io","gvk":"apiextensions.k8s.io/v1, Kind=CustomResourceDefinition","name":"addresspools.metallb.io","gvk":"apiextensions.k8s.io/v1, Kind=CustomResourceDefinition"}
2022-07-21T23:19:41.330959340Z {"level":"info","ts":1658445581.3308156,"logger":"cert-rotation","msg":"Ensuring CA cert","name":"bgppeers.metallb.io","gvk":"apiextensions.k8s.io/v1, Kind=CustomResourceDefinition","name":"bgppeers.metallb.io","gvk":"apiextensions.k8s.io/v1, Kind=CustomResourceDefinition"}
2022-07-21T23:19:41.442519108Z {"level":"info","ts":1658445581.4397717,"logger":"cert-rotation","msg":"Ensuring CA cert","name":"metallb-webhook-configuration","gvk":"admissionregistration.k8s.io/v1, Kind=ValidatingWebhookConfiguration","name":"metallb-webhook-configuration","gvk":"admissionregistration.k8s.io/v1, Kind=ValidatingWebhookConfiguration"}
2022-07-21T23:19:41.469917351Z {"level":"info","ts":1658445581.4690363,"logger":"cert-rotation","msg":"Ensuring CA cert","name":"addresspools.metallb.io","gvk":"apiextensions.k8s.io/v1, Kind=CustomResourceDefinition","name":"addresspools.metallb.io","gvk":"apiextensions.k8s.io/v1, Kind=CustomResourceDefinition"}
2022-07-21T23:19:41.499971745Z {"level":"info","ts":1658445581.4990368,"logger":"cert-rotation","msg":"Ensuring CA cert","name":"bgppeers.metallb.io","gvk":"apiextensions.k8s.io/v1, Kind=CustomResourceDefinition","name":"bgppeers.metallb.io","gvk":"apiextensions.k8s.io/v1, Kind=CustomResourceDefinition"}
2022-07-21T23:19:36.870435678Z {"caller":"config_controller.go:48","controller":"ConfigReconciler","level":"info","start reconcile":"/win-ap-kub01","ts":"2022-07-21T23:19:36Z"}
2022-07-21T23:19:36.870472577Z {"caller":"config_controller.go:136","controller":"ConfigReconciler","event":"force service reload","level":"info","ts":"2022-07-21T23:19:36Z"}
2022-07-21T23:19:36.870488196Z {"caller":"config_controller.go:147","controller":"ConfigReconciler","event":"config reloaded","level":"info","ts":"2022-07-21T23:19:36Z"}
2022-07-21T23:19:36.870493216Z {"caller":"config_controller.go:148","controller":"ConfigReconciler","end reconcile":"/win-ap-kub01","level":"info","ts":"2022-07-21T23:19:36Z"}
2022-07-21T23:19:36.870500229Z {"caller":"service_controller_reload.go:61","controller":"ServiceReconciler - reprocessAll","level":"info","start reconcile":"metallbreload/reload","ts":"2022-07-21T23:19:36Z"}
2022-07-21T23:19:36.870505949Z {"caller":"main.go:344","event":"serviceAnnounced","ips":["10.5.54.24"],"level":"info","msg":"service has IP, announcing","pool":"default","protocol":"layer2","ts":"2022-07-21T23:19:36Z"}
2022-07-21T23:19:36.872656091Z {"caller":"main.go:344","event":"serviceAnnounced","ips":["10.5.54.24"],"level":"info","msg":"service has IP, announcing","pool":"default","protocol":"layer2","ts":"2022-07-21T23:19:36Z"}
2022-07-21T23:19:36.872670488Z {"caller":"main.go:344","event":"serviceAnnounced","ips":["10.5.54.24"],"level":"info","msg":"service has IP, announcing","pool":"default","protocol":"layer2","ts":"2022-07-21T23:19:36Z"}
2022-07-21T23:19:36.872675327Z {"caller":"main.go:344","event":"serviceAnnounced","ips":["10.5.54.24"],"level":"info","msg":"service has IP, announcing","pool":"default","protocol":"layer2","ts":"2022-07-21T23:19:36Z"}
2022-07-21T23:19:36.872691618Z {"caller":"main.go:344","event":"serviceAnnounced","ips":["10.5.54.24"],"level":"info","msg":"service has IP, announcing","pool":"default","protocol":"layer2","ts":"2022-07-21T23:19:36Z"}
2022-07-21T23:19:36.872695996Z {"caller":"main.go:344","event":"serviceAnnounced","ips":["10.5.54.24"],"level":"info","msg":"service has IP, announcing","pool":"default","protocol":"layer2","ts":"2022-07-21T23:19:36Z"}
2022-07-21T23:19:36.872700705Z {"caller":"service_controller_reload.go:103","controller":"ServiceReconciler - reprocessAll","end reconcile":"metallbreload/reload","level":"info","ts":"2022-07-21T23:19:36Z"}
2022-07-21T23:19:41.593076923Z {"caller":"node_controller.go:42","controller":"NodeReconciler","level":"info","start reconcile":"/win-ap-kub01","ts":"2022-07-21T23:19:41Z"}
2022-07-21T23:19:41.593132337Z {"caller":"speakerlist.go:271","level":"info","msg":"triggering discovery","op":"memberDiscovery","ts":"2022-07-21T23:19:41Z"}
2022-07-21T23:19:41.593137908Z {"caller":"node_controller.go:64","controller":"NodeReconciler","end reconcile":"/win-ap-kub01","level":"info","ts":"2022-07-21T23:19:41Z"}
2022-07-21T23:19:41.593142396Z {"caller":"memberlist.go:268","component":"Memberlist","level":"warn","msg":"memberlist: Failed to resolve : lookup : no such host","ts":"2022-07-21T23:19:41Z"}
2022-07-21T23:19:41.593147035Z {"caller":"speakerlist.go:256","error":"1 error occurred:\n\t* Failed to resolve : lookup : no such host\n\n","expected":1,"joined":0,"level":"error","msg":"partial join","op":"memberDiscovery","ts":"2022-07-21T23:19:41Z"}
2022-07-21T23:19:41.593150802Z {"caller":"config_controller.go:48","controller":"ConfigReconciler","level":"info","start reconcile":"/win-ap-kub01","ts":"2022-07-21T23:19:41Z"}
2022-07-21T23:19:41.593154419Z {"caller":"config_controller.go:136","controller":"ConfigReconciler","event":"force service reload","level":"info","ts":"2022-07-21T23:19:41Z"}
2022-07-21T23:19:41.593158016Z {"caller":"config_controller.go:147","controller":"ConfigReconciler","event":"config reloaded","level":"info","ts":"2022-07-21T23:19:41Z"}
2022-07-21T23:19:41.593161652Z {"caller":"config_controller.go:148","controller":"ConfigReconciler","end reconcile":"/win-ap-kub01","level":"info","ts":"2022-07-21T23:19:41Z"}
2022-07-21T23:19:41.593168996Z {"caller":"service_controller_reload.go:61","controller":"ServiceReconciler - reprocessAll","level":"info","start reconcile":"metallbreload/reload","ts":"2022-07-21T23:19:41Z"}
2022-07-21T23:19:41.593174406Z {"caller":"main.go:344","event":"serviceAnnounced","ips":["10.5.54.26"],"level":"info","msg":"service has IP, announcing","pool":"default","protocol":"layer2","ts":"2022-07-21T23:19:41Z"}
2022-07-21T23:19:41.593178173Z {"caller":"service_controller_reload.go:103","controller":"ServiceReconciler - reprocessAll","end reconcile":"metallbreload/reload","level":"info","ts":"2022-07-21T23:19:41Z"}
2022-07-21T23:19:43.014632045Z {"caller":"memberlist.go:268","component":"Memberlist","level":"warn","msg":"memberlist: Failed to resolve : lookup : no such host","ts":"2022-07-21T23:19:43Z"}
2022-07-21T23:19:43.014673853Z {"caller":"speakerlist.go:256","error":"1 error occurred:\n\t* Failed to resolve : lookup : no such host\n\n","expected":1,"joined":0,"level":"error","msg":"partial join","op":"memberDiscovery","ts":"2022-07-21T23:19:43Z"}
2022-07-21T23:20:34.898940995Z {"caller":"memberlist.go:268","component":"Memberlist","level":"warn","msg":"memberlist: Failed to resolve : lookup : no such host","ts":"2022-07-21T23:20:34Z"}
2022-07-21T23:20:34.899149205Z {"caller":"speakerlist.go:256","error":"1 error occurred:\n\t* Failed to resolve : lookup : no such host\n\n","expected":1,"joined":0,"level":"error","msg":"partial join","op":"memberDiscovery","ts":"2022-07-21T23:20:34Z"}
```

3. If you're reporting a bug, please specify all of the following:
     - The bug itself, as detailed as you can.
     Metallb periodically reloads, dropping connections to all services for a short period.
     - Version of MetalLB
        0.13.4
     - Version of Kubernetes
         v1.23.8+k3s2
     - Name and version of network addon (e.g. Calico, Weave...)
         Cilium 1.12.0
     - Whether you've configured kube-proxy for iptables or ipvs mode
         - running cilium with ebpf using the kube-proxy replacement
         
This is all running on Ubuntu 22.04 server, installed on ESXi, running k3s.

**Comment by alphabet5** [comment]
I'm seeing the same thing on a brand new cluster, same specs as before with 0.13.4.

```
2022-08-02T00:03:51.712747131Z {"caller":"memberlist.go:268","component":"Memberlist","level":"warn","msg":"memberlist: Failed to resolve : lookup : no such host","ts":"2022-08-02T00:03:51Z"}
2022-08-02T00:03:51.712779256Z {"caller":"memberlist.go:268","component":"Memberlist","level":"warn","msg":"memberlist: Failed to resolve : lookup : no such host","ts":"2022-08-02T00:03:51Z"}
2022-08-02T00:03:51.712783061Z {"caller":"speakerlist.go:256","error":"2 errors occurred:\n\t* Failed to resolve : lookup : no such host\n\t* Failed to resolve : lookup : no such host\n\n","expected":2,"joined":0,"level":"error","msg":"partial join","op":"memberDiscovery","ts":"2022-08-02T00:03:51Z"}
2022-08-02T00:04:11.217500082Z {"caller":"node_controller.go:42","controller":"NodeReconciler","level":"info","start reconcile":"/ce-ap-k8s03","ts":"2022-08-02T00:04:11Z"}
2022-08-02T00:04:11.217545551Z {"caller":"speakerlist.go:271","level":"info","msg":"triggering discovery","op":"memberDiscovery","ts":"2022-08-02T00:04:11Z"}
2022-08-02T00:04:11.217576418Z {"caller":"node_controller.go:64","controller":"NodeReconciler","end reconcile":"/ce-ap-k8s03","level":"info","ts":"2022-08-02T00:04:11Z"}
2022-08-02T00:04:11.217728128Z {"caller":"memberlist.go:268","component":"Memberlist","level":"warn","msg":"memberlist: Failed to resolve : lookup : no such host","ts":"2022-08-02T00:04:11Z"}
2022-08-02T00:04:11.217817806Z {"caller":"memberlist.go:268","component":"Memberlist","level":"warn","msg":"memberlist: Failed to resolve : lookup : no such host","ts":"2022-08-02T00:04:11Z"}
2022-08-02T00:04:11.217877006Z {"caller":"speakerlist.go:256","error":"2 errors occurred:\n\t* Failed to resolve : lookup : no such host\n\t* Failed to resolve : lookup : no such host\n\n","expected":2,"joined":0,"level":"error","msg":"partial join","op":"memberDiscovery","ts":"2022-08-02T00:04:11Z"}
2022-08-02T00:04:11.217989236Z {"caller":"config_controller.go:48","controller":"ConfigReconciler","level":"info","start reconcile":"/ce-ap-k8s03","ts":"2022-08-02T00:04:11Z"}
2022-08-02T00:04:11.218466356Z {"caller":"config_controller.go:136","controller":"ConfigReconciler","event":"force service reload","level":"info","ts":"2022-08-02T00:04:11Z"}
2022-08-02T00:04:11.218526573Z {"caller":"config_controller.go:147","controller":"ConfigReconciler","event":"config reloaded","level":"info","ts":"2022-08-02T00:04:11Z"}
2022-08-02T00:04:11.218577902Z {"caller":"config_controller.go:148","controller":"ConfigReconciler","end reconcile":"/ce-ap-k8s03","level":"info","ts":"2022-08-02T00:04:11Z"}
2022-08-02T00:04:11.218690457Z {"caller":"service_controller_reload.go:61","controller":"ServiceReconciler - reprocessAll","level":"info","start reconcile":"metallbreload/reload","ts":"2022-08-02T00:04:11Z"}
2022-08-02T00:04:11.219096456Z {"caller":"main.go:344","event":"serviceAnnounced","ips":["172.22.111.214"],"level":"info","msg":"service has IP, announcing","pool":"default","protocol":"layer2","ts":"2022-08-02T00:04:11Z"}
2022-08-02T00:04:11.219851019Z {"caller":"main.go:344","event":"serviceAnnounced","ips":["172.22.111.214"],"level":"info","msg":"service has IP, announcing","pool":"default","protocol":"layer2","ts":"2022-08-02T00:04:11Z"}
2022-08-02T00:04:11.220114266Z {"caller":"service_controller_reload.go:103","controller":"ServiceReconciler - reprocessAll","end reconcile":"metallbreload/reload","level":"info","ts":"2022-08-02T00:04:11Z"}
2022-08-02T00:04:43.336804962Z {"caller":"config_controller.go:48","controller":"ConfigReconciler","level":"info","start reconcile":"/ce-ap-k8s02","ts":"2022-08-02T00:04:43Z"}
2022-08-02T00:04:43.336839105Z {"caller":"config_controller.go:136","controller":"ConfigReconciler","event":"force service reload","level":"info","ts":"2022-08-02T00:04:43Z"}
2022-08-02T00:04:43.336842065Z {"caller":"config_controller.go:147","controller":"ConfigReconciler","event":"config reloaded","level":"info","ts":"2022-08-02T00:04:43Z"}
2022-08-02T00:04:43.336850187Z {"caller":"config_controller.go:148","controller":"ConfigReconciler","end reconcile":"/ce-ap-k8s02","level":"info","ts":"2022-08-02T00:04:43Z"}
2022-08-02T00:04:43.336856982Z {"caller":"service_controller_reload.go:61","controller":"ServiceReconciler - reprocessAll","level":"info","start reconcile":"metallbreload/reload","ts":"2022-08-02T00:04:43Z"}
2022-08-02T00:04:43.336860847Z {"caller":"main.go:344","event":"serviceAnnounced","ips":["172.22.111.214"],"level":"info","msg":"service has IP, announcing","pool":"default","protocol":"layer2","ts":"2022-08-02T00:04:43Z"}
2022-08-02T00:04:43.336863237Z {"caller":"main.go:344","event":"serviceAnnounced","ips":["172.22.111.214"],"level":"info","msg":"service has IP, announcing","pool":"default","protocol":"layer2","ts":"2022-08-02T00:04:43Z"}
2022-08-02T00:04:43.336865715Z {"caller":"service_controller_reload.go:103","controller":"ServiceReconciler - reprocessAll","end reconcile":"metallbreload/reload","level":"info","ts":"2022-08-02T00:04:43Z"}
2022-08-02T00:04:51.712291903Z {"caller":"memberlist.go:268","component":"Memberlist","level":"warn","msg":"memberlist: Failed to resolve : lookup : no such host","ts":"2022-08-02T00:04:51Z"}
2022-08-02T00:04:51.712318170Z {"caller":"memberlist.go:268","component":"Memberlist","level":"warn","msg":"memberlist: Failed to resolve : lookup : no such host","ts":"2022-08-02T00:04:51Z"}
2022-08-02T00:04:51.712322052Z {"caller":"speakerlist.go:256","error":"2 errors occurred:\n\t* Failed to resolve : lookup : no such host\n\t* Failed to resolve : lookup : no such host\n\n","expected":2,"joined":0,"level":"error","msg":"partial join","op":"memberDiscovery","ts":"2022-08-02T00:04:51Z"}
```

**Comment by fedepaol** [comment]
This 
```2022-07-21T23:19:43.014632045Z {"caller":"memberlist.go:268","component":"Memberlist","level":"warn","msg":"memberlist: Failed to resolve : lookup : no such host","ts":"2022-07-21T23:19:43Z"}```

is pretty strange, because it should get an ip. The list of neighbors is built using status.podip of the speakers:
https://github.com/metallb/metallb/blob/c792677f195f7d9253d550eac963c4b53d9dc6a9/internal/k8s/k8s.go#L442

which would be strange if it was empty. 

I can craft an image with more logs, to see if it helps finding the issue.

**Comment by alphabet5** [comment]
@fedepaol So I think I just figured this out. I was installing metallb with helm, as well as applying the operator manifest… so that’s what was breaking things. Testing for 15m now and it's been fine.

I haven't figured out exactly what was changing or why it was breaking, but I don't think it really matters.

Trying to enable debug logging and it wasn't getting enabled then I discovered it was getting overwritten by the operator.
